Source/SwiftLintBuiltInRules/Rules/Lint/UnhandledThrowingTaskRule.swift

```diff
@@ -0,0 +1,238 @@
+                print(e)
+              }
+            }
+            """),
+            Example("""
+            ↓Task {
+              do {
+                throw FooError.bar
+              }
+            }
+            """),
+            Example("""
+            ↓Task {
+              throw FooError.bar
+            }
+            """),
+            Example("""
+            ↓Task<_, _> {
+              throw FooError.bar
+            }
+            """),
+            Example("""
+            ↓Task<Void,_> {
+              throw FooError.bar
+            }
+            """)
+        ]
+    )
+
+    func makeVisitor(file: SwiftLintFile) -> ViolationsSyntaxVisitor {
+        Visitor(viewMode: .sourceAccurate)
+    }
+}
+
+private extension UnhandledThrowingTaskRule {
+    final class Visitor: ViolationsSyntaxVisitor {
+        override func visitPost(_ node: FunctionCallExprSyntax) {
+            if node.hasViolation {
+                violations.append(node.calledExpression.positionAfterSkippingLeadingTrivia)
+            }
+        }
+    }
+}
+
+private extension FunctionCallExprSyntax {
+    var hasViolation: Bool {
+        isTaskWithImplicitErrorType &&
+            doesThrow &&
+            !(isAssigned || isValueAccessed)
+    }
+
+    var isTaskWithImplicitErrorType: Bool {
+        if let typeIdentifier = calledExpression.as(IdentifierExprSyntax.self),
+           typeIdentifier.identifier.text == "Task" {
+            return true
+        }
+
+        if let specializedExpression = calledExpression.as(SpecializeExprSyntax.self),
+           let typeIdentifier = specializedExpression.expression.as(IdentifierExprSyntax.self),
+           typeIdentifier.identifier.text == "Task",
+           let lastGeneric = specializedExpression.genericArgumentClause
+            .arguments.last?.argumentType.as(SimpleTypeIdentifierSyntax.self),
+           lastGeneric.typeName == "_" {
+            return true
+        }
+
+        return false
+    }
+
+    var isAssigned: Bool {
+        guard let parent else {
+            return false
+        }
+
+        if parent.is(InitializerClauseSyntax.self) {
+            return true
+        }
+
+        if let list = parent.as(ExprListSyntax.self),
+           list.contains(where: { $0.is(AssignmentExprSyntax.self) }) {
+            return true
+        }
+
+        return false
+    }
+
+    var isValueAccessed: Bool {
+        guard let parent = parent?.as(MemberAccessExprSyntax.self) else {
+            return false
+        }
+
+        return parent.name.text == "value"
+    }
+
+    var doesThrow: Bool {
+        ThrowsVisitor(viewMode: .sourceAccurate)
+            .walk(tree: self, handler: \.doesThrow)
+    }
+}
+
+private final class ThrowsVisitor: SyntaxVisitor {
+    var doesThrow = false
+
+    override func visit(_ node: DoStmtSyntax) -> SyntaxVisitorContinueKind {
+        // If there are no catch clauses, visit children to see if there are any try expressions.
```

Comment:
Does this handle these cases?

```swift
Task {
  do {
    try foo()
  } catch {
    try bar()
  }
}
```

or:

```swift
Task {
  do {
    try foo()
  } catch {
    throw BarError()
  }
}
```

Reply:
It did not, but with my latest changes it does, good call out!